tests/gctests.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+// Define objects and functions that j2me.js expects
+// but are unavailable in the shell environment.
+
+if (typeof console === "undefined") {
+  var console = {
+    log: print,
+  }
+}
+
+console.info = function (c) {
+  putstr(String.fromCharCode(c));
+};
+
+console.error = function (c) {
+  putstr(String.fromCharCode(c));
+};
+
+var freedAddr = 0;
+
+var passed = 0, failed = 0;
+
+function is(a, b, msg) {
+  if (a == b) {
+    ++passed;
+    print("pass " + msg);
+  } else {
+    ++failed;
+    print("fail " + msg + "; expected " + JSON.stringify(b) + ", got " + JSON.stringify(a));
+  }
+}
+
+function isNot(a, b, msg) {
+  if (a != b) {
+    ++passed;
+    print("pass " + msg);
+  } else {
+    ++failed;
+    print("fail " + msg + "; expected " + JSON.stringify(b) + " != " + JSON.stringify(a));
+  }
+}
+
+function ok(a, msg) {
+  if (!!a) {
+    ++passed;
+    print("pass " + msg);
+  } else {
+    ++failed;
+    print("fail " + msg);
+  }
+}
+
+var tests = [];
+
+function next() {
+  for (var i = 0; i < 100; i++) {
```

Comment:
I added this loop to make the tests as independent as possible of each other, but I'll remove it now because I think it's not enough.